src/redisjson.rs

```diff
@@ -45,23 +48,56 @@ impl From<Error> for redismodule::RedisError {
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub enum Format {
+    JSON,
+    BSON,
+}
+
+impl Format {
+    pub fn from_str(s: &str) -> Result<Format, Error> {
+        match s {
+            "JSON" => Ok(Format::JSON),
+            "BSON" => Ok(Format::BSON),
+            _ => return Err("ERR wrong format".into()),
+        }
+    }
+}
+
 #[derive(Debug)]
 pub struct RedisJSON {
     data: Value,
 }
 
 impl RedisJSON {
-    pub fn from_str(data: &str) -> Result<Self, Error> {
-        // Parse the string of data into serde_json::Value.
-        let v: Value = serde_json::from_str(data)?;
-
-        Ok(Self { data: v })
+    pub fn parse_str(data: &str, format: Format) -> Result<Value, Error> {
+        let value: Value = match format {
```

Comment:
Instead of the `match`, it would be idiomatic to use standard combinators like `map` and `map_err` (like we do in many other places).

Reply:
but we need map_or_else here and it seems like it's not stable yet

Reply:
You can use map and unwrap_or_else after it.

Reply:
done